code/modules/vending/_vending.dm

```diff
@@ -36,6 +36,8 @@
 	var/age_restricted = FALSE
 	///Whether the product can be recolored by the GAGS system
 	var/colorable
+	///List of items that have been returned to the vending machine.
+	var/list/returned_products = list()
```

Comment:
You should implement this as a lazylist, I took a count of /datum/data/vending_product datums on tram near roundstart and there were 3240 of them, which means a lot of empty and unused lists.